x-pack/filebeat/input/s3/collector.go

```diff
@@ -238,10 +238,14 @@ func getRegionFromQueueURL(queueURL string) (string, error) {
 
 // handle message
 func (c *s3Collector) handleSQSMessage(m sqs.Message) ([]s3Info, error) {
-	msg := sqsMessage{}
+	var msg sqsMessage
 	err := json.Unmarshal([]byte(*m.Body), &msg)
 	if err != nil {
-		return nil, fmt.Errorf("json unmarshal sqs message body failed: %w", err)
+		if jsonError, ok := err.(*json.SyntaxError); ok {
+			return nil, fmt.Errorf("json unmarshal sqs message body failed at offset %d with syntax error: %w", jsonError.Offset, err)
```

Comment:
@exekias Basically this PR is to enrich the error message a little bit more. Do you think it will actually help? Thanks!!

Reply:
Sounds good! Will it be possible to use the given offset somehow? I wonder if we should debug the full message in these cases

Reply:
Yep, done! Thanks!